### PR TITLE
Add warning messages in console for deprecated parameters

### PR DESCRIPTION
**Pull request by ikasarov**
Parameters now refactored to enum;
Added check for deprecated parameters; custom messages for parameters

#### Description: 
Changed SupportedParameters as an enum type - now supporting name + additional fileds;
Added logic for cli warning if using deprecated descriptor parameters.

#### Issue: 
JIRA 1073

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationsCloudModelBuilder.java [hunk 1]
```diff
@@ -135,15 +140,56 @@ protected CloudApplicationExtended createCloudApplication(String name, String mo
...
+            if (userMessageLogger != null) {
+                userMessageLogger.info(Messages.GENERAL_DEPRECATED_PARAMETER, combinedParameterNames);
+                if (combinedCustomDeprecationMessage != null) {
+                    userMessageLogger.info(combinedCustomDeprecationMessage);
+                }
+            }
+        }
+    }
+
+    protected List<Parameter> filterDeprecatedParameters(Collection<String> parameterNames) {
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationsCloudModelBuilder.java]: `filter` is a bit ambiguous. Does it return the deprecated or the non-deprecated parameters?

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/Parameter.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+    private static void initIsDeprecated() {
+        for (Parameter param : Parameter.values()) {
+            try {
+                Field field = Parameter.class.getField(param.toString());
+                if (field.isAnnotationPresent(Deprecated.class)) {
+                    param.isDeprecated = true;
+                }
+            } catch (NoSuchFieldException e) {
+                e.printStackTrace();
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/Parameter.java]: Handle the exception in some way. Maybe throw an IllegalStateException.

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/Parameter.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            } catch (SecurityException e) {
+                e.printStackTrace();
```
Comment (nictas) [inline com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/model/Parameter.java]: Handle both exceptions in the same catch block by using `|`.

File: com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationsCloudModelBuilder.java [hunk 4]
```diff
@@ -135,15 +140,56 @@ protected CloudApplicationExtended createCloudApplication(String name, String mo
...
-        applicationEnvCloudModelBuilder.removeSpecialServiceProperties(merged);
-        for (String parameterName : merged.keySet()) {
+
+        warnAboutDeprecatedParameters(merged);
+        warnAboutUnsupportedParameters(merged);
+    }
+
+    protected void warnAboutUnsupportedParameters(Map<String, Object> parametersInApp) {
+        applicationEnvCloudModelBuilder.removeSpecialApplicationProperties(parametersInApp);
+        applicationEnvCloudModelBuilder.removeSpecialServiceProperties(parametersInApp);
```
[the remark this answers is not in the document]
Comment (ikasarov) [reply com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/v2/ApplicationsCloudModelBuilder.java]: This is legacy logic - only formal changes here.

**Pull request merged**